[6.0] Numbers custom fields plugin

TLWebdesign opened the pull request:
I'm trying to create a new custom fields plugin for the standard Joomla Form Field Number. We have it for Integer and i'd like to have the same for the number field. That way w'll also gain ability to use floats. 
 
### Summary of Changes
Extra plugin added for the numbers standard form field type. With added currency options for display. Making it very useful for display of menu's for restaurants or anything else where a currency or a number is needed to be displayed.

### Testing Instructions
You can discover the plugin and then use it in custom fields. It's called numbers. 
Create a custom field of type numbers and check weather the options available do what they supposed to be doing. especially related to the number formatting.

### Actual result BEFORE applying this Pull Request
No numbers custom field available.

### Expected result AFTER applying this Pull Request
Numbers custom field available and working.


### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

TLWebdesign committed:
First draft for numbers custom field plugin

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+; Note : All ini files need to be saved as UTF-8
+
+PLG_FIELDS_NUMBER="Fields - Number"
+PLG_FIELDS_INTEGER_LABEL="Number (%s)"
```
```suggestion
PLG_FIELDS_NUMBER_LABEL="Number (%s)"
```

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+
+PLG_FIELDS_NUMBER="Fields - Number"
+PLG_FIELDS_INTEGER_LABEL="Number (%s)"
+PLG_FIELDS_INTEGER_PARAMS_FIRST_LABEL="Min"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_FIRST_LABEL="Min"
```

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+PLG_FIELDS_NUMBER="Fields - Number"
+PLG_FIELDS_INTEGER_LABEL="Number (%s)"
+PLG_FIELDS_INTEGER_PARAMS_FIRST_LABEL="Min"
+PLG_FIELDS_INTEGER_PARAMS_LAST_LABEL="Max"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_LAST_LABEL="Max"
```

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+PLG_FIELDS_INTEGER_LABEL="Number (%s)"
+PLG_FIELDS_INTEGER_PARAMS_FIRST_LABEL="Min"
+PLG_FIELDS_INTEGER_PARAMS_LAST_LABEL="Max"
+PLG_FIELDS_INTEGER_PARAMS_MULTIPLE_LABEL="Multiple"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_MULTIPLE_LABEL="Multiple"
```

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+PLG_FIELDS_INTEGER_PARAMS_FIRST_LABEL="Min"
+PLG_FIELDS_INTEGER_PARAMS_LAST_LABEL="Max"
+PLG_FIELDS_INTEGER_PARAMS_MULTIPLE_LABEL="Multiple"
+PLG_FIELDS_INTEGER_PARAMS_STEP_LABEL="Step Increment"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_STEP_LABEL="Step Increment"
```

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,12 @@
+PLG_FIELDS_INTEGER_PARAMS_LAST_LABEL="Max"
+PLG_FIELDS_INTEGER_PARAMS_MULTIPLE_LABEL="Multiple"
+PLG_FIELDS_INTEGER_PARAMS_STEP_LABEL="Step Increment"
+PLG_FIELDS_INTEGER_XML_DESCRIPTION="This plugin lets you create new fields of type 'number' in any extensions where custom fields are supported."
```
```suggestion
PLG_FIELDS_NUMBER_XML_DESCRIPTION="This plugin lets you create new fields of type 'number' in any extensions where custom fields are supported."
```

brianteeman commented on plugins/fields/number/number.xml:
```diff
@@ -0,0 +1,63 @@
+					default="1.0"
+					filter="float"
+				/>
+
```
```suggestion
```

brianteeman commented on plugins/fields/number/number.xml:
```diff
@@ -0,0 +1,63 @@
+					default="100.0"
+					filter="float"
+				/>
+
```
```suggestion
```

brianteeman commented on plugins/fields/number/services/provider.php:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @return  void
+     *
+     * @since   4.3.0
```
```suggestion
     * @since   __DEPLOY_VERSION__
```

brianteeman commented on plugins/fields/number/src/Extension/Number.php:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Fields Number Plugin
+ *
+ * @since  5.3.0
```
```suggestion
 * @since  __DEPLOY_VERSION__
```

TLWebdesign committed:
Apply suggestions from code review

Co-authored-by: Brian Teeman <brian@teeman.net>

TLWebdesign commented:
@brianteeman Thanks for reviewing and fixing my mistakes.

brianteeman commented on plugins/fields/number/src/Extension/Number.php:
```diff
@@ -0,0 +1,27 @@
+ * @package     Joomla.Plugin
+ * @subpackage  Fields.number
+ *
+ * @copyright   (C) 2017 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on plugins/fields/number/services/provider.php:
```diff
@@ -0,0 +1,46 @@
+ * @package     Joomla.Plugin
+ * @subpackage  Fields.number
+ *
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on plugins/fields/number/tmpl/number.php:
```diff
@@ -0,0 +1,22 @@
+ * @package     Joomla.Plugin
+ * @subpackage  Fields.Number
+ *
+ * @copyright   (C) 2016 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```

TLWebdesign committed:
Apply suggestions from code review

Co-authored-by: Brian Teeman <brian@teeman.net>

TLWebdesign commented:
So the wall i'm hitting now is that i can't figure out how the display of the custom field in an "article edit view" is set or adjusted. Looking at for instance the integer custom field it sort of automagically seems to work. Because i can't seem to find a reference anywhere in the plugin folder itself. 

Could someone point me in the right direction here?

Quy commented on plugins/fields/number/tmpl/number.php:
```diff
@@ -0,0 +1,22 @@
+if (is_numeric($value)) {
+    $value = (float) $value;
+} else {
+    $value = '';
+    $value = isset($min) ? $min : $value;
```
```suggestion
    $value = $min ?? '';
```

sandewt commented:
> I'm not there yet and i need help!

There is still some work to be done regarding the databases and the installation, see e.g.:

https://github.com/TLWebdesign/joomla-cms/blob/f9dec464683f19d5efe9699ee8745d703cea7250/installation/sql/mysql/base.sql#L296

A nice example of another plugin can be found here:
[https://github.com/joomla/joomla-cms/pull/42777/files](https://github.com/joomla/joomla-cms/pull/42777/files)

HLeithner commented:
This pull request has been automatically rebased to 5.3-dev.

TLWebdesign committed:
added install sql and removed unused field param

suvarnakale commented:
I have tested this item :red_circle: unsuccessfully on 3d8f2bc98eda99bad39f434aba21c0f8ce8f10ba<br /><br />I have tested this item and it is not working as expected. 

For more information - On discover number field plugin is available and installable. But after onstallation, in standard joomla form the number field is not available. 

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

shindebalu commented:
I have tested this item :red_circle: unsuccessfully on 3d8f2bc98eda99bad39f434aba21c0f8ce8f10ba<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

fgsw commented:
@shindebalu Can you please provide the resons for your test result? Also notice that this Pull Request is a draft.

HLeithner commented:
This pull request has been automatically rebased to 6.0-dev.

TLWebdesign committed:
Merge branch '6.0-dev' into 5.2-dev

TLWebdesign committed:
updated copyright year

TLWebdesign committed:
Update plugins/fields/number/tmpl/number.php

Co-authored-by: Quy <quy@nomonkeybiz.com.unlinked202412130709>

TLWebdesign commented:
I think this one is ready for review. I was wondering tho;

Does anyone think it would be a good idea to implement more layouts or an option to format the number like for instance a currency? Or would that be too much and we should stick to default behaviour of the same core field type? 

I'm asking because i have a use case where i want to use this field to display prices. Would be nice if i could set the number formatting in the field settings.

TLWebdesign commented:
> like the integer field? 

No i was thinking using https://www.php.net/manual/en/numberformatter.formatcurrency.php and either get locale from the default language or have a list of locales to select from in a dropdown list? The last one would be the "lightest" on rendering is my guess. you could also add an option for the number of fraction digits.

Or just a regular number_format and have 3 settings for currency, thousand seperator and decimal operator.

TLWebdesign committed:
Added update sql and currency formatting option

I've added update sql and also implemented a currency formatting option to display the number as a currency.

TLWebdesign commented on installation/sql/postgresql/base.sql:
@brianteeman Should i reorder the rest of the plugins on new install or should i add it to the bottom? I'm also doing other plugins as you read on mattermost and now it's conflicting with the order. What is the preferred way to deal with this?

TLWebdesign replied:
@QuyTon could you maybe answer above question or tag someone who could answer this?

TLWebdesign replied:
Got the info from brian so i'm closing this conversation as resolved and will update the pr as per the best practice.

TLWebdesign committed:
cleaned up language strings

Removed unused language strings and changed mistakes and mismatches in naming. Also improved some text

TLWebdesign committed:
Fixed years to 2025 and version number to be in line with joomla version

fgsw commented on administrator/language/en-GB/plg_fields_number.ini:
Please alphasort language constants.

fgsw reviewed (changes requested):
Please alphasort language constants.

fgsw commented on administrator/language/en-GB/plg_fields_number.ini:
Please alphasort language constants.

TLWebdesign replied:
OK i think i did it correctly now.

TLWebdesign committed:
Alpha sorted the strings

TLWebdesign committed:
fixed stupid mistake.

TLWebdesign committed:
updated sql files to include all the default params settings

QuyTon commented on plugins/fields/number/number.xml:
```diff
@@ -0,0 +1,102 @@
+					layout="joomla.form.field.radio.switcher"
+					default="0"
+					filter="integer"
+				>
```
```suggestion
					>
```
Please repeat with other instances.

TLWebdesign replied:
Thanks for taking time to review. I changed all occurences like you mentioned

TLWebdesign committed:
changed indentation

QuyTon commented on plugins/fields/number/params/number.xml:
```diff
@@ -0,0 +1,77 @@
+			</field>
+
+			<field
+				name="decimals"
```
Please sort in this order: name, type, label, description

TLWebdesign replied:
Ok, done! Thanks

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,24 @@
+
+PLG_FIELDS_NUMBER="Fields - Number"
+PLG_FIELDS_NUMBER_LABEL="Number (%s)"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_DESC="Format the number as currency?"
```
The description is basically the same without additional info as the label so I would remove it.

TLWebdesign replied:
OK i removed it.

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,24 @@
+PLG_FIELDS_NUMBER="Fields - Number"
+PLG_FIELDS_NUMBER_LABEL="Number (%s)"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_DESC="Format the number as currency?"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_LABEL="Format as currency"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_CURRENCY_LABEL="Format as Currency"
```

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,24 @@
+PLG_FIELDS_NUMBER_LABEL="Number (%s)"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_DESC="Format the number as currency?"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_LABEL="Format as currency"
+PLG_FIELDS_NUMBER_PARAMS_DECIMALS_DESC="Number of decimals (0-10)"
+PLG_FIELDS_NUMBER_PARAMS_DECIMALS_LABEL="Decimals"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_DECIMALS_LABEL="Number of Decimals"
```

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,24 @@
+PLG_FIELDS_NUMBER_PARAMS_POSITION_OPTION_BEFORE="Before"
+PLG_FIELDS_NUMBER_PARAMS_STEP_DESC="If user click up/down arrow current value will change by this attribute (this attribute value will be added or subtracted to current field value)"
+PLG_FIELDS_NUMBER_PARAMS_STEP_LABEL="Step Increment"
+PLG_FIELDS_NUMBER_PARAMS_SYMBOL_DESC="Symbol to put in front or after the number"
```
This description may not be necessary.

TLWebdesign replied:
Which one/ I see 2 descriptions there line 20 or 22?

QuyTon replied:
PLG_FIELDS_NUMBER_PARAMS_POSITION_DESC
PLG_FIELDS_NUMBER_PARAMS_SYMBOL_DESC

TLWebdesign committed:
fix sorting of field params

TLWebdesign committed:
Update administrator/language/en-GB/plg_fields_number.ini

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

TLWebdesign committed:
Update administrator/language/en-GB/plg_fields_number.ini

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

TLWebdesign committed:
removed unnecessary description

TLWebdesign committed:
Changed update SQL file to own file and reverted changes to other file

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,22 @@
+PLG_FIELDS_NUMBER_LABEL="Number (%s)"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_LABEL="Format as Currency"
+PLG_FIELDS_NUMBER_PARAMS_DECIMALS_LABEL="Number of Decimals"
+PLG_FIELDS_NUMBER_PARAMS_MAX_DESC="This value is the highest that can be chosen (only limits the up/down arrows)"
+PLG_FIELDS_NUMBER_PARAMS_MAX_LABEL="Max"
+PLG_FIELDS_NUMBER_PARAMS_MIN_DESC="This value is the lowest that can be chosen (only limits the up/down arrows)"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_MAX_DESC="Maximum value that can be chosen (only limits the up/down arrows)."
PLG_FIELDS_NUMBER_PARAMS_MAX_LABEL="Max"
PLG_FIELDS_NUMBER_PARAMS_MIN_DESC="Minimum value that can be chosen (only limits the up/down arrows)."
```

QuyTon commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,22 @@
+PLG_FIELDS_NUMBER_LABEL="Number (%s)"
+PLG_FIELDS_NUMBER_PARAMS_CURRENCY_LABEL="Format as Currency"
+PLG_FIELDS_NUMBER_PARAMS_DECIMALS_LABEL="Number of Decimals"
+PLG_FIELDS_NUMBER_PARAMS_MAX_DESC="This value is the highest that can be chosen (only limits the up/down arrows)"
+PLG_FIELDS_NUMBER_PARAMS_MAX_LABEL="Max"
+PLG_FIELDS_NUMBER_PARAMS_MIN_DESC="This value is the lowest that can be chosen (only limits the up/down arrows)"
+PLG_FIELDS_NUMBER_PARAMS_MIN_LABEL="Min"
```
```suggestion
PLG_FIELDS_NUMBER_PARAMS_MAX_DESC="Maximum value that can be chosen (only limits the up/down arrows)."
PLG_FIELDS_NUMBER_PARAMS_MAX_LABEL="Maximum"
PLG_FIELDS_NUMBER_PARAMS_MIN_DESC="Minimum value that can be chosen (only limits the up/down arrows)."
PLG_FIELDS_NUMBER_PARAMS_MIN_LABEL="Minimum"
```

TLWebdesign committed:
Update administrator/language/en-GB/plg_fields_number.ini

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

TLWebdesign committed:
Merge branch '6.0-dev' into 5.2-dev

TLWebdesign committed:
Merge branch '6.0-dev' into 5.2-dev

TLWebdesign committed:
Merge branch '6.0-dev' into 5.2-dev

Bodge-IT reviewed (approved):
Github shows 1 change request still outstanding but I've checked all these and they all seem resolved.

Bodge-IT committed:
Merge branch '6.0-dev' into 5.2-dev

brianteeman commented:
Please review all the _desc strings to see if they are really needed. 

For example isnt this just obvious from the label and the field options
PLG_FIELDS_NUMBER_PARAMS_POSITION_DESC="Place the symbol before or after number"

TLWebdesign committed:
removed unnecessary descriptions

TLWebdesign committed:
removed unneccessary descriptions

TLWebdesign commented:
> @TLWebdesign - I tested this PR, it needs at least 1 fix before I can mark it successful.
> 
> When Toggling in the back-end the Toggle Inline Help, the last field Number of Decimals has a language-string instead of text: PLG_FIELDS_NUMBER_PARAMS_DECIMALS_DESC
> 
> ALSO, seeing that the minimum has no bearing on Saving a field with a lower number, it would be good to update the text of the Minimum field: Minimum value that can be chosen (only limits the up/down arrows). to be clearer...
> 
> Something like: User can enter any value and a minimum value that can be chosen that limits the up/down arrows. (or whatever you prefer)
> 
> When I set my Minimum to 100000, and saved my field with 5000, it let me save with no warning (even though the minimum was 100000) but then when I used the arrows, it jumped to 100000 - this is perhaps intended but needs clearer info.

The missing language string is fixed. the desc shouldn't be on this field but it was left in the form definition. 

For the language, it already states this in the description the same way as the official documentation does. So i'd like to just keep it consistent with that.

exlemor commented:
I have tested this item :white_check_mark: successfully on b89199b5bdb6a61a1a36d7521e411b72f1ebf515<br /><br />I have tested this successfully! Thanks @TLWebdesign for the fixes.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

brianteeman commented on administrator/language/en-GB/plg_fields_number.ini:
```diff
@@ -0,0 +1,20 @@
+PLG_FIELDS_NUMBER_PARAMS_POSITION_LABEL="Symbol Position"
+PLG_FIELDS_NUMBER_PARAMS_POSITION_OPTION_AFTER="After"
+PLG_FIELDS_NUMBER_PARAMS_POSITION_OPTION_BEFORE="Before"
+PLG_FIELDS_NUMBER_PARAMS_STEP_DESC="If a user click up/down arrow current value will change by this attribute (this attribute value will be added or subtracted to current field value)"
```
is this really needed?

If you think it is then please follow the guide and end the sentence with a period

TLWebdesign replied:
Hmm figured it would not hurt. But really needed. probably not, so i removed it 👍

TLWebdesign committed:
removed more unnecessary descriptions and added default in the params file

exlemor commented:
I have tested this item :white_check_mark: successfully on 8e8f3e2c1c2e0054047e9f85a74e69c209fc52b4<br /><br />I have re-tested this successfully. Thanks @TLWebdesign <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

LadySolveig commented:
I have tested this item :white_check_mark: successfully on 8e8f3e2c1c2e0054047e9f85a74e69c209fc52b4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

richard67 commented:
@TLWebdesign Could you rename the update SQL scripts to something newer than "6.0.0-2025-08-02.sql", e.g. "6.0.0-2025-08-17.sql? Thanks in advance.

richard67 commented:
P.S. If the name of the new update SQL script is not newer than the latest present one when compared with PHP version compare, it will not run when updating e.g. from 6.0.0-alpha3.

richard67 commented on administrator/components/com_admin/sql/updates/mysql/6.0.0-2024-08-24.sql:
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`, `client_id`, `enabled`, `access`, `protected`, `locked`, `manifest_cache`, `params`, `custom_data`, `ordering`, `state`) VALUES
+(0, 'plg_fields_number', 'plugin', 'number', 'fields', 0, 1, 1, 0, 1, '', '{"min":"1.0","max":"100.0","step":"0.1","currency":"0","position":"0","decimals":"2"}', '', 17, 0);
```
```suggestion
(0, 'plg_fields_number', 'plugin', 'number', 'fields', 0, 1, 1, 0, 1, '', '{"min":"1.0","max":"100.0","step":"0.1","currency":"0","position":"0","decimals":"2"}', '', -1, 0);
```
New plugins should be installed on update with order number -1 so they are shown at the top.

richard67 replied:
And the script has to be renamed, see my other comment in the main thread.

TLWebdesign replied:
ok name is changed. Thansk for looking at this.

richard67 replied:
No, the name is not changes. Please check again it is 2024!!!

TLWebdesign replied:
Sorry thought i pushed the change. At the ER with my kid atm waiting. Things are a bit hectic.

richard67 commented on administrator/components/com_admin/sql/updates/postgresql/6.0.0-2024-08-24.sql:
```diff
@@ -0,0 +1,2 @@
+INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder", "client_id", "enabled", "access", "protected", "locked", "manifest_cache", "params", "custom_data", "ordering", "state") VALUES
+(0, 'plg_fields_number', 'plugin', 'number', 'fields', 0, 1, 1, 0, 1, '', '{"min":"1.0","max":"100.0","step":"0.1","currency":"0","position":"0","decimals":"2"}', '', 17, 0);
```
```suggestion
(0, 'plg_fields_number', 'plugin', 'number', 'fields', 0, 1, 1, 0, 1, '', '{"min":"1.0","max":"100.0","step":"0.1","currency":"0","position":"0","decimals":"2"}', '', -1, 0);
```
New plugins should be installed on update with order number -1 so they are shown at the top.

TLWebdesign committed:
Update administrator/components/com_admin/sql/updates/postgresql/6.0.0-2024-08-24.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

TLWebdesign committed:
changed sql name

TLWebdesign committed:
Merge branch '5.2-dev' of https://github.com/TLWebdesign/joomla-cms into numbers-custom-field

TLWebdesign committed:
Update 6.0.0-2024-08-24.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

exlemor commented:
I have tested this item :white_check_mark: successfully on 3044c1576f08bd473c38b3b23ac6818c65904688<br /><br />I have re-tested this successfully. Thanks @TLWebdesign <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43974">issues.joomla.org/tracker/joomla-cms/43974</a>.</sub>

richard67 commented:
Sorry, wrong button.

richard67 commented:
@exlemor Have you also tested updating from 6.0.0-alpha3? Then you would have noticed that the field will not be added on update due to the wrong update SQL script name.

For PRs which add new update SQL scripts updating should always be part of the test.

@TLWebdesign Maybe you should update the testing instructions accordingly.

richard67 commented:
Filenames "6.0.0-2024-08-24.sql" are still wrong. Please rename to "6.0.0-2025-08-17.sql". Mind also the year!

Bodge-IT committed:
Rename 6.0.0-2024-08-24.sql to 6.0.0-2025-08-17.sql

Bodge-IT committed:
Rename 6.0.0-2024-08-24.sql to 6.0.0-2025-08-17.sql

TLWebdesign committed:
Merge branch '5.2-dev' of https://github.com/TLWebdesign/joomla-cms into numbers-custom-field

TLWebdesign commented:
> Filenames "6.0.0-2024-08-24.sql" are still wrong. Please rename to "6.0.0-2025-08-17.sql". Mind also the year!

sorry in the rush to get going here i didn't wait long enough for my laptop to commit the renaming. but i see it is done now. Thanks for your help

richard67 commented:
I will set RTC as it had 2 tests.

@Bodge-IT @softforge When this and the other one with the other field gets merged before beta 1, then we (or you) have to test if updating still works on MySQL and PostgreSQL. I will try to find time for that tomorrow morning, at least for the PostgreSQL part.

richard67 committed:
Merge branch '6.0-dev' into 5.2-dev

richard67 committed:
Merge branch '6.0-dev' into 5.2-dev

Bodge-IT committed:
Merge branch '6.0-dev' into 5.2-dev

Bodge-IT commented:
Thanks @TLWebdesign, well done.  Thanks for all the tests
